container/rkt/handler.go

```diff
@@ -0,0 +1,488 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Handler for "rkt" containers.
+package rkt
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/google/cadvisor/container"
+	"github.com/google/cadvisor/container/common"
+	"github.com/google/cadvisor/container/libcontainer"
+	"github.com/google/cadvisor/fs"
+	info "github.com/google/cadvisor/info/v1"
+	"github.com/google/cadvisor/utils"
+
+	"github.com/golang/glog"
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+	cgroupfs "github.com/opencontainers/runc/libcontainer/cgroups/fs"
+	"github.com/opencontainers/runc/libcontainer/configs"
+	"golang.org/x/exp/inotify"
+)
+
+type rktContainerHandler struct {
+	// Name of the container for this handler.
+	name               string
+	cgroupSubsystems   *libcontainer.CgroupSubsystems
+	machineInfoFactory info.MachineInfoFactory
+
+	// Inotify event watcher.
+	watcher *common.InotifyWatcher
+
+	// Signal for watcher thread to stop.
+	stopWatcher chan error
+
+	// Absolute path to the cgroup hierarchies of this container.
+	// (e.g.: "cpu" -> "/sys/fs/cgroup/cpu/test")
+	cgroupPaths map[string]string
+
+	// Manager of this container's cgroups.
+	cgroupManager cgroups.Manager
+
+	// Whether this container has network isolation enabled.
+	hasNetwork bool
+
+	fsInfo         fs.FsInfo
+	externalMounts []common.Mount
+
+	rootFs string
+
+	isPod bool
+
+	aliases []string
+
+	pid int
+
+	rootfsStorageDir string
+
+	// Filesystem handler.
+	fsHandler common.FsHandler
+
+	ignoreMetrics container.MetricSet
+}
+
+func (self *rktContainerHandler) GetCgroupPaths() map[string]string {
+	return self.cgroupPaths
+}
+
+func (self *rktContainerHandler) GetMachineInfoFactory() info.MachineInfoFactory {
+	return self.machineInfoFactory
+}
+
+func (self *rktContainerHandler) GetName() string {
+	return self.name
+}
+
+func (self *rktContainerHandler) GetExternalMounts() []common.Mount {
+	return self.externalMounts
+}
+
+func (self *rktContainerHandler) HasNetwork() bool {
+	return self.hasNetwork && !self.ignoreMetrics.Has(container.NetworkUsageMetrics)
+}
+
+func (self *rktContainerHandler) HasFilesystem() bool {
+	if !self.ignoreMetrics.Has(container.DiskUsageMetrics) {
+		return true
+	}
+	return false
+}
+
+func newRktContainerHandler(name string, cgroupSubsystems *libcontainer.CgroupSubsystems, machineInfoFactory info.MachineInfoFactory, fsInfo fs.FsInfo, watcher *common.InotifyWatcher, rootFs string, ignoreMetrics container.MetricSet) (container.ContainerHandler, error) {
+	rktPath := "/var/lib/rkt"
```

Comment:
It will be.  At this point I just wanted to get it in wit ha static path, we've dded support for getting this info out of the api service, just need to figure out how to use its api.